Composer/packages/client/src/components/GetStarted/GetStartedNextSteps.tsx

```diff
@@ -0,0 +1,223 @@
+import { dispatcherState, settingsState } from '../../recoilModel';
+import { mergePropertiesManagedByRootBot } from '../../recoilModel/dispatchers/utils/project';
+import { rootBotProjectIdSelector } from '../../recoilModel/selectors/project';
+import { navigateTo } from '../../utils/navigation';
+
+import { GetStartedTask } from './GetStartedTask';
+import { h3Style } from './styles';
+
+type GetStartedProps = {
+  requiresLUIS: boolean;
+  requiresQNA: boolean;
+};
+
+export type NextSteps = {
+  checked: boolean;
+  key: string;
+  description: string;
+  learnMore: string;
+  required?: boolean;
+  label: string;
+  onClick: (step?: NextSteps) => void;
+};
+
+export const GetStartedNextSteps: React.FC<GetStartedProps> = (props) => {
+  const projectId = useRecoilValue(currentProjectIdState);
+  const botProjects = useRecoilValue(localBotsDataSelector);
+  const botProject = botProjects.find((b) => b.projectId === projectId);
+  const [displayManageLuis, setDisplayManageLuis] = useState<boolean>(false);
+  const [displayManageQNA, setDisplayManageQNA] = useState<boolean>(false);
+
+  const { setSettings } = useRecoilValue(dispatcherState);
+  const rootBotProjectId = useRecoilValue(rootBotProjectIdSelector) || '';
+  const settings = useRecoilValue(settingsState(projectId));
+  const mergedSettings = mergePropertiesManagedByRootBot(projectId, rootBotProjectId, settings);
+  const [requiredNextSteps, setNextSteps] = useState<NextSteps[]>([]);
+  const [optionalSteps, setOptionalSteps] = useState<NextSteps[]>([]);
+
+  const hideManageLuis = () => {
+    setDisplayManageLuis(false);
+  };
+  const hideManageQNA = () => {
+    setDisplayManageQNA(false);
+  };
+
+  const openLink = (link) => {
+    TelemetryClient.track('GettingStartedLinkClicked', { method: 'link', url: link });
+    navigateTo(link);
+  };
+
+  const doNextStep = (currentStep) => {
+    const nextAction = requiredNextSteps.filter((f) => !f.checked && f.key != currentStep)?.[0];
+    if (nextAction) {
+      nextAction.onClick();
+    }
+  };
+
+  const updateLuisSettings = (newLuisSettings) => {
+    setSettings(projectId, {
+      ...mergedSettings,
+      luis: { ...mergedSettings.luis, ...newLuisSettings },
+    });
+  };
+
+  const updateQNASettings = (newQNASettings) => {
+    setSettings(projectId, {
+      ...mergedSettings,
+      qna: { ...mergedSettings.qna, ...newQNASettings },
+    });
+  };
+
+  const linkToPackageManager = `/bot/${projectId}/plugin/package-manager/package-manager`;
+  const linkToConnections = `/bot/${projectId}/botProjectsSettings/#connections`;
+  const linkToLUISSettings = `/bot/${projectId}/botProjectsSettings/#luisKey`;
+  const linktoQNASettings = `/bot/${projectId}/botProjectsSettings/#qnaKey`;
+  const linkToLGEditor = `/bot/${projectId}/language-generation`;
+  const linkToLUEditor = `/bot/${projectId}/language-understanding`;
+
+  useEffect(() => {
+    const newNextSteps: NextSteps[] = [];
+
+    let hasLUIS = false;
```

Comment:
Nit: You could write this as direct assignments to hasLuis and hasQna.  This would be 2 lines rather than 8.